### DISCUSSION TITLE
Bing Maps is deprecated, will Azure Maps work with Mapsui instead?

### DISCUSSION THREAD

**Discussion by giobiondani** [Mapsui/Mapsui#2801] [Ideas]
I noticed here https://www.bingmapsportal.com/ a communication that Bing Maps is deprecated and will be replaced by Azure Maps.
Here there are details concerning the migration path: https://learn.microsoft.com/en-us/azure/azure-maps/migrate-bing-maps-overview

I haven't looked into the details yet, but I assume that this will affect Mapsui as well?

**Comment by pauldendulk** [comment]
I need to look into this if I find time. Usually this kind of thing is possible if you set the right headers. In BruTile v6 you use the configureHttpRequestMessage on the HttpTileSource for this.

**Comment by giobiondani** [comment]
This was actually a lot less complicated than I thought.

All I had to do was create a HttpTileSource using this URL:
`https://atlas.microsoft.com/map/tile?api-version=2024-04-01&tilesetId=microsoft.imagery&zoom={z}&x={x}&y={y}&subscription-key=" + the_key`
In this example the tileset is the satellite imagery.

The other tilesets that are available can be found at this page: https://learn.microsoft.com/en-us/rest/api/maps/render/get-map-tileset?view=rest-maps-2025-01-01&tabs=HTTP

**Reply by pauldendulk** [reply]
Thanks for the feedback. This will help others!
